deploy-nagios/tasks/main.yml

```diff
@@ -0,0 +1,101 @@
+  register: create_project_result
+  failed_when: create_project_result.stderr and create_project_result.stderr != '' and 'already exists' not in create_project_result.stderr
+  changed_when: create_project_result.rc == 0 or (create_project_result == 1 and 'created' in create_project_result.stdout)
+
+-
+  name: "Create Nagios serviceaccount"
+  command: "oc create serviceaccount nagios --namespace {{ project_name }}"
+  register: nagios_sa_result
+  failed_when:
+    - nagios_sa_result.stderr != '' and 'already exists' not in nagios_sa_result.stderr
+    - "'created' not in nagios_sa_result.stdout"
+  changed_when: nagios_sa_result.rc == 0
+
+-
+  name: "Read list of service accounts for this namepsace"
+  command: oc get policyBindings :default -n {{ project_name }} --template {% raw %}"{{(index .roleBindings 0).roleBinding.subjects}}"{% endraw %}
+  register: sa_list
+  failed_when: False
+  changed_when: False
+
+-
+  name: "Apply admin role to Nagios user"
+  command: "oc policy add-role-to-user admin -z nagios  --namespace {{ project_name }} "
+  register: nagios_role_result
+  changed_when: "'nagios' not in sa_list.stdout and nagios_role_result.rc == 0"
+
+- 
+  name: "Create the Nagios resources"
+  command: "oc new-app --namespace {{ project_name }} -f {{ buildfarm_templates_dir }}/nagios-template.json"
+  register: nagios_create_result
+  failed_when:
+    - nagios_create_result.stderr and nagios_create_result.stderr != '' and 'already exists' not in nagios_create_result.stderr
+  changed_when: nagios_create_result.rc == 0 or (nagios_create_result.rc == 1 and 'created' in nagios_create_result.stdout)
+-
+  name: "Poll for Pods to become ready"
+  command: oc get pods --namespace={{ project_name }} --output jsonpath='{.items[*].status.conditions[?(@.type=="Ready")].status}'
+  register: poll_result
+  until: "'False' not in poll_result.stdout"
+  retries: 60
+  delay: 10
+  when: nagios_create_result|changed
+
+- name: Force status check to schedule in Nagios container
+  shell: "oc exec --namespace {{ project_name }} $(oc get po --namespace {{ project_name }}  | grep nagios | awk '{print $1}') -c nagios -- python /opt/rhmap/host-svc-check"
+  register: nagios_result
+  failed_when: False
+
+
+- name: Execute status check in Nagios container
+  shell: "oc exec  --namespace {{ project_name }}  $(oc get po --namespace {{ project_name }}  | grep nagios | awk '{print $1}') -c nagios -- python /opt/rhmap/check-status | \
+          sed -e 's/16/critical/' -e 's/1/pending/' -e 's/2/healthy/' -e 's/4/warning/' -e 's/8/unknown/' "
+  register: nagios_result
+  failed_when: False
+
+
+-
+  name: "Verify Nagios dashboard is green, fail otherwise"
+  shell: echo {{ item }}
+  register: nagios_dashboard
+  with_items: "{{ nagios_result.stdout_lines }}"
+  when:
+    - (mongo_replicas is defined and mongo_replicas == 3) or (role is defined and role == "core") or (role is defined and role == "dedicated")
```

Comment:
We use this check in a couple of places (mbaas, core) etc. at the end of the deployment.
The `when` logic here is incorrect though. I can fix that logic and keep the task or remove it if you think it's unnecessary. 
It checks if there are some critical results on any of the nagios checks and will fail the playbook if there is. Although this is the last task so it could remove the fail clause and we could just output the result of the nagios checks